### DISCUSSION TITLE
My cobbler installation doesn't seem to work...

### DISCUSSION THREAD

**Discussion by 13126690189** [cobbler/cobbler#2782] [Q&A]
I didn't run 'cobbler get-loaders'. Actually, the command doesn't work for me. It just reports a 404 error..
I used the script mkgrub.sh to generate the loaders.

After everything is ready, I start a new virtual machine and configure the BIOS to boot from network.
The virtual machine successfully got an IP via DHCP, and downloaded the file 'grub/grub.0' (I know this through tcpdump), then it blocked, without showing up any menus, or printing any messages.

I tried modifying /etc/dhcp/dhcpd.conf manually, changing 'filename' from 'grub/grub.0' to 'pxelinux.0'. This time the virtual machine shows up the cobbler menu as expected.

Why doesn't grub.0 work？Thanks!

**Comment by SchoolGuy** [comment]
What bootloader did you set for the system? In older version you can only set one and in newer you can set multiple ones. If you don't enable the generation for the corresponding bootloader indeed nothing is happening.

**Reply by 13126690189** [reply]
I didn't add a system. I thought a new machine would get the filename of the correct bootloader based on the architecture it reports in its DHCP request...Actually, I don't know how a new machine, which is not added as a system, fetch the filename of the correct bootloader to use..

I am quite new to cobbler, Maybe my question is too naive. Thank you for your pacience :-)

**Reply by SchoolGuy** [reply]
Okay no problem. Do you have a distribution and a profile added in Cobbler? If yes then after a Cobbler `sync` this should be enough. If you look in the generated `dhcpd.conf` you will see the entry and how Cobbler decides what `filename` in tftp is used.

**Reply by iuvooneill** [reply]
I am having the same problem with non-EFI boots...I'm setting up a new cobbler server on Rocky 8.5. I've got everything set up. I see it download grub/grub.0, but grub.cfg never gets requested after. The system will sit for a while, then eventually give a grub shell.

In the case of an EFI boot, it works fine...grub/grubx64.efi loads, and then it searches for the config eventually finding grub.cfg and presenting the menu.

In both cases, the bootloader was created with mkgrub.sh.
